### PR TITLE
hareThirdParty.hare-ssh: init at unstable-2023-11-16

### PR DESCRIPTION
**Pull request by patwid**
## Description of changes

[hare-ssh](https://git.sr.ht/~sircmpwn/hare-ssh) package provides an implementation of the SSH client & server protocol for Hare.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by qbit** [comment]
Result of `nixpkgs-review pr 276437` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>hareThirdParty.hare-ssh</li>
  </ul>
</details>

**Review by qbit — APPROVED** [review]

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -14128,6 +14128,12 @@
     githubId = 15645854;
     name = "Brad Christensen";
   };
+  patwid = {
+    email = "patrick.widmer@tbwnet.ch";
+    github = "patwid";
+    githubId = 25278658;
+    name = "Patrick Widmer";
+  };
```
Comment (Janik-Haag) [inline maintainers/maintainer-list.nix:14136]: please split this into a separate commit called `maintainers: add patwid`

**Review by Janik-Haag — COMMENTED** [review]
Please rebase and add the suggestions, should be good afterwards.

**Comment by patwid** [comment]
> Please rebase and add the suggestions, should be good afterwards.

Done

File: pkgs/development/hare-third-party/hare-ssh/default.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+    export HARECACHE="$NIX_BUILD_TOP/.harecache"
+    export BINOUT="$NIX_BUILD_TOP/.bin"
+
+    makeFlagsArray+=(
+      PREFIX="${builtins.placeholder "out"}"
+    )
+
+    runHook postConfigure
+  '';
```
Comment (starzation) [inline pkgs/development/hare-third-party/hare-ssh/default.nix:27]: Maybe my fault that doing this hard way, anyway we can use `makeFlags` insted.

```suggestion
  makeFlags = [
    "PREFIX=${builtins.placeholder "out"}"
    "HARECACHE=.harecache"
  ];
```

File: pkgs/development/hare-third-party/hare-ssh/default.nix [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+  doCheck = true;
+
+  meta = with lib; {
+    homepage = "https://git.sr.ht/~sircmpwn/hare-ssh/";
+    description = "SSH client & server protocol implementation for Hare";
+    license = with licenses; [ mpl20 ];
+    maintainers = with maintainers; [ patwid ];
+
+    inherit (harec.meta) platforms badPlatforms;
```
Comment (starzation) [inline pkgs/development/hare-third-party/hare-ssh/default.nix:37]: Remove the use of `harec`

```suggestion
    inherit (hare.meta) platforms badPlatforms;
```

File: pkgs/development/hare-third-party/hare-ssh/default.nix [hunk 4]
```diff
@@ -0,0 +1,39 @@
+{ lib, stdenv, hare, harec, hareThirdParty, fetchFromSourcehut }:
```
Comment (starzation) [inline pkgs/development/hare-third-party/hare-ssh/default.nix:1]:
```suggestion
{ lib, stdenv, hare, hareThirdParty, fetchFromSourcehut }:
```

**Review by starzation — CHANGES_REQUESTED** [review]

File: pkgs/development/hare-third-party/hare-ssh/default.nix (hunk 4 above)
Comment (patwid) [reply pkgs/development/hare-third-party/hare-ssh/default.nix]: Applied

File: pkgs/development/hare-third-party/hare-ssh/default.nix (hunk 3 above)
Comment (patwid) [reply pkgs/development/hare-third-party/hare-ssh/default.nix]: Applied

File: pkgs/development/hare-third-party/hare-ssh/default.nix (hunk 2 above)
Comment (patwid) [reply pkgs/development/hare-third-party/hare-ssh/default.nix]: Applied

**Comment by onemoresuza** [comment]
Result of `nixpkgs-review pr 276437` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>hareThirdParty.hare-ssh</li>
  </ul>
</details>

**Review by qbit — APPROVED** [review]

**Review by Janik-Haag — APPROVED** [review]
